test/integration/http-integration-test.js

```diff
@@ -0,0 +1,254 @@
+
+const TIMEOUT = 20000;   // how long before each test case times out
+
+const apiOptions = {
+  server: 'wss://s1.ripple.com'
+};
+
+const httpPort = 3000;
+
+function createClient() {
+  return jayson.client.http({port: httpPort, hostname: 'localhost'});
+}
+
+const address = 'r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59';
+
+describe('http server integration tests', function() {
+  this.timeout(TIMEOUT);
+
+  let server = null;
+
+  beforeEach(function(done) {
+    server = createHTTPServer(apiOptions, httpPort);
+    server.start().then(() => {
+      this.client = createClient();
+      done();
+    });
+  });
+
+  afterEach(function(done) {
+    server.stop().then(done);
+  });
+
+  it('getLedgerVersion', function(done) {
+    this.client.request('getLedgerVersion', [], (err, result) => {
+      if (err) {
+        return done(err);
+      }
+      assert(_.isNumber(result.result));
+      done();
+    });
+  });
+
+  it('getServerInfo', function(done) {
+    this.client.request('getServerInfo', [], (err, result) => {
+      if (err) {
+        return done(err);
+      }
+      assert(_.isNumber(result.result.validatedLedger.ledgerVersion));
+      done();
+    });
+  });
+
+  it('getServerInfo - named params', function(done) {
+    this.client.request('getServerInfo', {}, (err, result) => {
+      if (err) {
+        return done(err);
+      }
+      assert(_.isNumber(result.result.validatedLedger.ledgerVersion));
+      done();
+    });
+  });
+
+  it('getTransaction', function(done) {
+    const id =
+      '4EB6B76237DEEE99F1EA16FAACED2D1E69C5F9CB54F727A4ECA51A08AD3AF466';
+    this.client.request('getTransaction', [id], '2', (err, result) => {
+      if (err) {
+        return done(err);
+      }
+      assert.deepEqual(result, fixtures.getTransaction);
+      done();
+    });
+  });
+
+  it('getTransaction - named params', function(done) {
+    const id =
+      '4EB6B76237DEEE99F1EA16FAACED2D1E69C5F9CB54F727A4ECA51A08AD3AF466';
+    this.client.request('getTransaction', {id: id}, '2', (err, result) => {
+      if (err) {
+        return done(err);
+      }
+      assert.deepEqual(result, fixtures.getTransaction);
+      done();
+    });
+  });
+
+  it('getTransactions', function(done) {
+    const account = 'rpP2JgiMyTF5jR5hLG3xHCPi1knBb1v9cM';
+    const options = {
+      binary: true,
+      limit: 1,
+      start: 'FBAAC31D6BAEEFA9E501266FD62DA7A7982662BC19BC42F49BB41405C2F820DB'
+    };
+
+    this.client.request('getTransactions', {address: account, options}, '3',
+    (err, result) => {
+      if (err) {
+        return done(err);
+      }
+      assert.deepEqual(result, fixtures.getTransactions);
+      done();
+    });
+  });
+
+  it('prepareSettings', function(done) {
+    const instructions = {
+      maxFee: '0.000012',
+      sequence: 23,
+      maxLedgerVersion: 8820051
+    };
+    const settings = apiRequests.prepareSettings;
+
+    this.client.request('prepareSettings',
+    [address, settings, instructions], '4',
```

Comment:
I think it's probably best to keep support for positional arguments since that seems to be common for JSON RPC. But I think we should document and use named parameters exclusively.